wolfcrypt/src/port/arm/armv8-chacha.c

```diff
@@ -0,0 +1,2397 @@
+            "EOR v1.16B, v1.16B, v29.16B \n\t"
+            "EOR v2.16B, v2.16B, v30.16B \n\t"
+            "EOR v3.16B, v3.16B, v31.16B \n\t"
+            "LD1 { v28.4S-v31.4S }, [%[m]] \n\t"
+            "MOV v27.S[0], w17 \n\t"
+            "ST1 { v0.4S-v3.4S }, [%[c]], #64 \n\t"
+
+            "ADD v4.4S, v4.4S, v24.4S \n\t"
+            "ADD v5.4S, v5.4S, v25.4S \n\t"
+            "ADD v6.4S, v6.4S, v26.4S \n\t"
+            "ADD v7.4S, v7.4S, v27.4S \n\t"
+            "EOR v4.16B, v4.16B, v28.16B \n\t"
+            "EOR v5.16B, v5.16B, v29.16B \n\t"
+            "EOR v6.16B, v6.16B, v30.16B \n\t"
+            "EOR v7.16B, v7.16B, v31.16B \n\t"
+            "ST1 { v4.4S-v7.4S }, [%[c]] \n\t"
+
+
+            : [c] "+r" (c), [m] "+r" (m)
+            : [rounds] "r" (ROUNDS/2), [input] "r" (input),
+              [chacha_chunk_bytes] "I" (CHACHA_CHUNK_BYTES)
+            : "memory", "cc",
+              "x17",
+              "v0",  "v1",  "v2",  "v3",  "v4",
+              "v5",  "v6",  "v7",
+              "v12", "v13",
+              "v24", "v25", "v26", "v27", "v28", "v29", "v30", "v31"
+    );
+#else
+    __asm__ __volatile__ (
+            "MOV r11, %[rounds] \n\t"
+            "MOV r12, #1 \n\t"
+            "VLDM %[input], { q0-q3 } \n\t"
+            "VMOV.I32 q8, #0 \n\t"
+            "VMOV q4, q0 \n\t"
+            "VMOV.I32 d16[0], r12 \n\t"
+            "VMOV q5, q1 \n\t"
+            "VMOV q6, q2 \n\t"
+            "VADD.I32 q7, q3, q8 \n\t" // add one to counter
+
+            // store input
+            "VMOV q10, q0 \n\t"
+            "VMOV q11, q1 \n\t"
+            "VMOV q12, q2 \n\t"
+            "VMOV q13, q3 \n\t"
+
+            "loop_128_%=: \n\t"
+            "SUBS r11, r11, #1 \n\t"
+
+            // ODD ROUND
+            "VADD.I32 q0, q0, q1 \n\t"
+            "VADD.I32 q4, q4, q5 \n\t"
+            "VEOR q8, q3, q0 \n\t"
+            "VEOR q9, q7, q4 \n\t"
+            // rotation by 16 bits may be done by reversing the 16 bit elements in 32 bit words
+            "VREV32.16 q3, q8 \n\t"
+            "VREV32.16 q7, q9 \n\t"
+
+            "VADD.I32 q2, q2, q3 \n\t"
+            "VADD.I32 q6, q6, q7 \n\t"
+            "VEOR q8, q1, q2 \n\t"
+            "VEOR q9, q5, q6 \n\t"
+            // SIMD instructions don't support rotation so we have to cheat using shifts and a help register
+            "VSHL.I32 q1, q8, #12 \n\t"
+            "VSHL.I32 q5, q9, #12 \n\t"
+            "VSRI.I32 q1, q8, #20 \n\t"
+            "VSRI.I32 q5, q9, #20 \n\t"
+
+            "VADD.I32 q0, q0, q1 \n\t"
+            "VADD.I32 q4, q4, q5 \n\t"
+            "VEOR q8, q3, q0 \n\t"
+            "VEOR q9, q7, q4 \n\t"
+            // SIMD instructions don't support rotation so we have to cheat using shifts and a help register
+            "VSHL.I32 q3, q8, #8 \n\t"
+            "VSHL.I32 q7, q9, #8 \n\t"
+            "VSRI.I32 q3, q8, #24 \n\t"
+            "VSRI.I32 q7, q9, #24 \n\t"
+
+            "VADD.I32 q2, q2, q3 \n\t"
+  
```

Comment:
Change to this code - that is calculate the number of bytes to process here and no need to return anything.

Reply:
Do you mean in the 320 byte block function?

Reply:
Weird???
Added a comment on a line in the wc_Chacha_encrypt function and it didn't show up properly.
I'm talking about the SPARKI code in that function.